### PR TITLE
Adding Endpoint type

### PR DESCRIPTION
**Pull request by hosseinsh**
Lemur endpoints can be of different types, and the certificates attached to them, can be stored in different certificate registries such as AWS IAM or AWS ACM, with a unique  path.

This changes modifies the endpoint table to contain two new columns and populates them during the endpoint sync job: 
- registry_type: e.g., `iam` or `acm`
- certificate_path: e.g., `/cloudfront/`

This allows Lemur to build more accurate ARNs, which allows better certificate rotation.

### CODE REVIEW

File: lemur/plugins/lemur_aws/iam.py [hunk 1]
```diff
@@ -38,24 +38,68 @@ def retry_throttled(exception):
...
+    'arn:aws:iam::123456789012:server-certificate/example.com'
+    'arn:aws:iam::123456789012:server-certificate/cloudfront/example.com-cloudfront'
+    'arn:aws:acm:us-west-2:123456789012:certificate/example.com'
+
 
     :param arn: IAM SSL arn
     :return: name of the certificate as uploaded to AWS
     """
-    return arn.split("/", 1)[1]
+    return arn.split("/")[-1]
```
Comment (hosseinsh) [inline lemur/plugins/lemur_aws/iam.py:50]: this allows the name to be without the path, as the path doesn't help with finding the certname in Lemur, as done in the next step

File: lemur/plugins/lemur_aws/iam.py [hunk 2]
```diff
@@ -38,24 +38,68 @@ def retry_throttled(exception):
...
     :param account_number:
     :param region:
     :param certificate_name:
     :return:
     """
-    return "arn:aws:iam::{account_number}:server-certificate/{certificate_name}".format(
-        account_number=account_number, certificate_name=certificate_name
-    )
+    if path == '':
+        return f"arn:aws:iam::{account_number}:server-certificate/{certificate_name}"
```
Comment (hosseinsh) [inline lemur/plugins/lemur_aws/iam.py:100]: default behavior

File: lemur/plugins/lemur_aws/plugin.py [hunk 3]
```diff
@@ -112,9 +112,9 @@ def get_elb_endpoints(account_number, region, elb_dict):
             dnsname=elb_dict["DNSName"],
             type="elb",
             port=listener["Listener"]["LoadBalancerPort"],
-            certificate_name=iam.get_name_from_arn(
-                listener["Listener"]["SSLCertificateId"]
-            ),
+            certificate_name=iam.get_name_from_arn(listener["Listener"]["SSLCertificateId"]),
+            certificate_path=iam.get_path_from_arn(listener["Listener"]["SSLCertificateId"]),
```
Comment (hosseinsh) [inline lemur/plugins/lemur_aws/plugin.py:116]: persisting the new information in the returned dict, and later into the endpoint table

File: lemur/plugins/lemur_aws/plugin.py [hunk 4]
```diff
@@ -267,7 +269,11 @@ def update_endpoint(self, endpoint, certificate):
 
         # relies on the fact that region is included in DNS name
         region = get_region_from_dns(endpoint.dnsname)
-        arn = iam.create_arn_from_cert(account_number, region, certificate.name)
+        if endpoint.registery_typ == 'iam':
+            arn = iam.create_arn_from_cert(account_number, region, certificate.name, endpoint.certificate_path)
```
Comment (hosseinsh) [inline lemur/plugins/lemur_aws/plugin.py:273]: a more accurate `arn`, should help with catching any corner cases, and help with a better understanding about all discovered endpoints.

File: lemur/migrations/versions/2201c548a5a1_.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+"""empty message
```
Comment (jtschladen) [inline lemur/migrations/versions/2201c548a5a1_.py:1]: Could you add a description here? I know the migration is fairly self-descriptive, but I think anything is better than `empty message` :)

**Review by jtschladen — APPROVED** [review]

File: lemur/migrations/versions/2201c548a5a1_.py (hunk 5 above)
Comment (hosseinsh) [reply lemur/migrations/versions/2201c548a5a1_.py:1]: excellent point. definitely plus one!

File: lemur/plugins/lemur_aws/iam.py [hunk 6]
```diff
@@ -39,23 +39,71 @@ def get_name_from_arn(arn):
     """
     Extract the certificate name from an arn.
 
-    :param arn: IAM SSL arn
+    examples:
+    'arn:aws:iam::123456789012:server-certificate/example.com'
+    'arn:aws:iam::123456789012:server-certificate/cloudfront/example.com-cloudfront'
+    'arn:aws:acm:us-west-2:123456789012:certificate/example.com'
```
[the remark this answers is not in the document]
Comment (hosseinsh) [reply lemur/plugins/lemur_aws/iam.py]: very good idea. makes it more convenient to see what the method is supposed to do, without the need to look at the code or the tests. 
addressed all the suggestions.

**Review by charhate — APPROVED** [review]

File: lemur/plugins/lemur_aws/plugin.py (hunk 4 above)
Comment (hosseinsh) [reply lemur/plugins/lemur_aws/plugin.py:273]: this also captures the registry_type` NoneType case.

**Pull request merged**